python_modules/dagster/dagster/_core/definitions/utils.py

```diff
@@ -226,7 +226,7 @@ def normalize_tags(
 
 # Inspired by allowed Kubernetes labels:
 # https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
-VALID_DEFINITION_TAG_KEY_REGEX_STR = r"^([A-Za-z0-9_.-]{1,63}/)?[A-Za-z0-9_.-]{1,63}$"
+VALID_DEFINITION_TAG_KEY_REGEX_STR = r"^([A-Za-z0-9_.-]{1,63}/|dagster/kind/)?[A-Za-z0-9_.-]{1,63}$"
```

Comment:
This is at a minimum worthy of a comment and also it seems suspect that `kind` is hardcoded here.

Reply:
Pulled out to a separate set of whitelisted nested namespaces + added test